### ISSUE TITLE
Mikkelsen normal maps

### ISSUE THREAD

**Issue by unicomp21** [rhulha/ChronosGL#26]
How hard would it be to add support for mikkt space bump maps?  These are pretty much the standard these days for xnormal, blender, ue4, unity3d, etc.  I've got some cool stuff I'd like to create demos for if we can get this added.  I did a pyramid demo a while back which consumed crazy amounts of bandwidth, this is the ultimate way to cut down on size while keeping detail.

three.js code is here:
https://threejs.org/examples/webgl_materials_bumpmap.html

**Comment by muth02446** [comment]
I do not think this is very hard and it would be nice to have this as an example.
The three js code uses a json file for the vertex data and a jpeg for the bump map.
It also references a zip archive with equivalent information  including and obj file -- and
ChronosGL already has a loader for obj files (c.f. obj example).

So what is missing is a new shader that takes the bump maps into account.
You could start with the shader used by the light or shadow examples.
I can do some handholding if necessary.

**Comment by muth02446** [comment]
I have just added a  new example that is about 90% there:

https://github.com/rhulha/ChronosGL/tree/master/example/bumpmap

**Comment by unicomp21** [comment]
Thanks!

**Comment by chick-fil-a-21** [comment]
Any updates?
